beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -997,11 +998,38 @@ func (s *Server) GetSyncCommitteeDuties(w http.ResponseWriter, r *http.Request)
 		pubkey48 := bytesutil.ToBytes48(pubkey)
 		committeePubkeys[pubkey48] = append(committeePubkeys[pubkey48], strconv.FormatUint(uint64(j), 10))
 	}
-	duties, err := syncCommitteeDuties(requestedValIndices, st, committeePubkeys)
+	duties, vals, err := syncCommitteeDutiesAndVals(requestedValIndices, st, committeePubkeys)
```

Comment:
it might more readable if this function was duplicated instead of having the if statements duplicated
or you can set a function inside to alias the function like 
```
var operation (s beaconState.BeaconState, epoch primitives.Epoch, pubKey []byte, status ethpb.ValidatorStatus) error

operation = core.RegisterSyncSubnetCurrentPeriod  or core.RegisterSyncSubnetNextPeriod
and use 

operation(st, requestedEpoch, pk[:], valStatus)
```
this way you can skip copying and pasting most of the loop or even put it in its own function.